HumanExposure/factotum
# 1893: functional use related audit log

1. Added update trigger to FunctionalUse table, when report_funcuse/category_id changes, add to auditlog table on ALL related chemicals, on `report_funcuse` and `harmonized category` (category title) fields if changed.
2. Added update trigger to FunctionalUseCategory table, when title changes, add to auditlog table on ALL related chemicals.
3. Added insert/update/delete trigger to FunctionalUseToRawChem table, when related report_funcuse/category title changes, add to auditlog table on associated chemical.
4. The above triggers does not really fit in the current general scheme. So add them manually.
5. Add migration to convert the audit on functional_use_id field from last sprint to report_funcuse/harmonized category fields.

To test:
Add/remove functional use to chemical, verify it is properly audited.
Change functional use reported field, verify related chemicals are properly audited.
Change functional use category title, verify related chemicals are properly audited.
